### ISSUE TITLE
BoringSSL conflicting symbols for Armv7

### ISSUE THREAD

**Issue by colemancda** [apple/swift-nio-ssl#407]
I encountered an issue with integrating CryptoKit (nested dependency) and SwiftNIO in a single project, since its statically linked and both seem to vendor BoringSSL, are there any plans to merge those? There are libraries that use CryptoKit on iOS and swift-crypto for Linux, that are gonna need workarounds to compile and link on Linux.
`swift-nio-ssl/CNIOBoringSSL`
`swift-crypto/CCryptoBoringSSL`
[Swift-armv7-CryptoKit.log](https://github.com/apple/swift-nio-ssl/files/9841516/Swift-armv7-CryptoKit.log)

**Comment by Lukasa** [comment]
We'll need to add support for armv7 builds to our cross-compile checks. Are you able to produce a x-compile toolchain for Linux armv7 we can use?

**Comment by colemancda** [comment]
I'm currently working on that, been having some issues.

**Comment by colemancda** [comment]
I was able to generate a toolchain, https://github.com/colemancda/swift-armv7/actions/runs/3304509240
Just clone https://github.com/colemancda/swift-armv7 and run 

```
export SWIFT_PACKAGE_SRCDIR=/home/user/Developer/MySwiftPackage
export SWIFT_NATIVE_PATH=/tmp/cross-toolchain/debian-bullseye.sdk
./generate-xcode-toolchain.sh
./build-swift-package.sh
```

**Comment by Lukasa** [comment]
So I think that doesn't quite have the shape we're looking for. Ideally it should produce a toolchain that we can use with `--destination` as shown here: https://github.com/apple/swift-nio-ssl/blob/4fb7ead803e38949eb1d6fabb849206a72c580f3/scripts/vendor-boringssl.sh#L114.

**Comment by colemancda** [comment]
I agree, but I adopted this from https://github.com/apple/swift-package-manager/blob/main/Utilities/build_ubuntu_cross_compilation_toolchain and wanted to make as few changes as possible

**Comment by colemancda** [comment]
Also I wasn't aware that the destination.json format would work without absolute paths.
